build/translations/csv-sync.js

```diff
@@ -1,37 +1,105 @@
 const fs = require('fs')
+const args = process.argv.slice(2)
 const PATH_CSV = './front-end/assets/translations/'
-const files = fs.readdirSync(PATH_CSV)
 const masterFile = 'en.csv'
-const MasterTranslations = fs.readFileSync(`${PATH_CSV}${masterFile}`, 'utf8').match(/[^\r\n]+/g)
-MasterTranslations.shift()
-function updateMissingTranslation (masterLine, slaveArray) {
-  const masterKey = masterLine.split(',')[0]
-  const sK = slaveArray.find(s => masterKey === s.split(',')[0])
+const files = fs.readdirSync(PATH_CSV).filter(f => f !== masterFile)
+const stringify = require('csv-stringify')
+const parse = require('csv-parse/lib/sync')
+
+function readMasterFile() {
+  return readTranslation(masterFile)
+}
+
+function readTranslation(filename) {
+  const source = fs.readFileSync(`${PATH_CSV}${filename}`, 'utf8')
+  const records = parse(source, {
+    columns: true,
+    skip_empty_lines: true
+  })
+  return records
+}
+
+function updateMissingTranslation(masterTranslation, slaveTranslationArray) {
+  const sK = slaveTranslationArray.find(s => masterTranslation.Key === s.Key)
   if (!sK) {
-    slaveArray.push(`${masterKey},`)
+    slaveTranslationArray.push({ Key: masterTranslation.Key, Translation: '' })
   }
 }
-function deleteDeprecatedTranslations (slaveLine, masterArray) {
-  const slaveKey = slaveLine.split(',')[0]
-  const found = masterArray.find(s => slaveKey === s.split(',')[0])
-  if (!found) {
-    console.log(`Deprecated key: ${slaveKey}`)
-    slaveLine = ''
-  }
-  return slaveLine
+
+function deleteDeprecatedTranslations(slaveArray, masterArray) {
+  const masterKeys = masterArray.map(x => x.Key)
+  slaveArray = slaveArray.filter(x => masterKeys.includes(x.Key))
+  return slaveArray
+}
+
+function syncTranslationFiles(masterTranslations) {
+  files.forEach(f => {
+    let slave = fs.readFileSync(`${PATH_CSV}${f}`, 'utf8')
```

Comment:
dialect, not slave

Reply:
so by this, i guess `master` should be renamed as `reference` ?